boat-maven-plugin/src/main/java/com/backbase/oss/boat/RadioMojo.java

```diff
@@ -0,0 +1,171 @@
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+@Mojo(name = "radio", requiresDependencyResolution = ResolutionScope.RUNTIME, threadSafe = true)
+@Slf4j
+@Getter
+@Setter
+public class RadioMojo extends AbstractMojo {
+
+    @Parameter(property = "groupId", defaultValue = "${project.groupId}", required = true)
+    private String groupId;
+
+    @Parameter(property = "artifactId", defaultValue = "${project.artifactId}", required = true)
+    private String artifactId;
+
+    @Parameter(property = "version", defaultValue = "${project.version}", required = true)
+    private String version;
+
+    @Parameter(property = "basePath", required = true)
+    private String basePath;
+
+    @Parameter(property = "portalKey", required = true)
+    private String portalKey;
+
+    @Parameter(property = "sourceKey", required = true)
+    private String sourceKey;
+
+    @Parameter(property = "username", required = false)
+    private String username;
+
+    @Parameter(property = "password", required = false)
+    private String password;
+
+    @Parameter(property = "specs")
+    private SpecConfig[] specs;
+
+    @Parameter(name = "radioOutput", defaultValue = "${project.build.directory}/target/boat-radio-report")
+    private File radioOutput;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+
+        BasicAuthRequestInterceptor basicAuthRequestInterceptor = null;
+
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        if(username != null && !username.isEmpty() && password != null && !password.isEmpty()){
+             basicAuthRequestInterceptor = new BasicAuthRequestInterceptor(username, password);
+        }else{
+            getLog().warn("No Authentication set");
+        }
+
+        List<UploadSpec> allSpecs = Arrays.stream(specs).map(this::mapToUploadSpec).collect(Collectors.toList());
+
+        ApiClient apiClient = new ApiClient().setBasePath(basePath);
+        if(basicAuthRequestInterceptor != null){
+            apiClient.addAuthorization("Basic Auth", basicAuthRequestInterceptor);
+        }
+
+        BoatMavenPluginApi api = apiClient.buildClient(BoatMavenPluginApi.class);
+
+        UploadRequestBody uploadRequestBody = new UploadRequestBody();
+        uploadRequestBody.setGroupId(groupId);
```

Comment:
A lot of plugins allow the use of <artifact> A string of the form groupId:artifactId:version[:packaging[:classifier]].

https://maven.apache.org/plugins/maven-dependency-plugin/get-mojo.html#artifact

Reply:
Contract is specified by boat-bay. So we need to change it there as well to a single string. Might as well change the data model to use this string instead of separate fields?

Reply:
That would have just been a convenience option for the mojo. Technically not important so no need to change anything.